ArduPlane/quadplane.cpp

```diff
@@ -1330,26 +1388,51 @@ bool QuadPlane::should_assist(float aspeed, bool have_airspeed)
 
     const uint32_t now = AP_HAL::millis();
 
+    const bool should_assist_alt = !in_vtol_land_sequence() && !in_vtol_mode();
+
     /*
       optional assistance when altitude is too close to the ground
      */
-    if (assist_alt > 0) {
+    if (should_assist_alt && assist_alt.alt > 0) {
+        float height_above_ground = plane.relative_ground_altitude(plane.g.rangefinder_landing);
+        if (height_above_ground < assist_alt.alt) {
+            if (assist_alt.start_ms == 0) {
+                assist_alt.start_ms = now;
+            }
+            if (now - assist_alt.start_ms > assist_delay*1000) {
+                // we've been below assistant alt for Q_ASSIST_DELAY seconds
+                if (!assist_alt.in_assist) {
+                    start_alt_assist(height_above_ground);
```

Comment:
This seems to be doing the assistance action based on the first assistance alt not the new one?